diff --git a/mkspecs/features/sis_targets.prf b/mkspecs/features/sis_targets.prf
index bcfdf8f..45ebe30 100644
--- a/mkspecs/features/sis_targets.prf
+++ b/mkspecs/features/sis_targets.prf
@@ -28,7 +28,7 @@ equals(GENERATE_SIS_TARGETS, true) {
                               )
 
         ok_sis_target.target = ok_sis
-        ok_sis_target.commands = createpackage.bat $(QT_SIS_OPTIONS) $$basename(TARGET)_template.pkg \
+        ok_sis_target.commands = createpackage.pl $(QT_SIS_OPTIONS) $$basename(TARGET)_template.pkg \
                                      $(QT_SIS_TARGET) $(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)
 
         unsigned_sis_target.target = unsigned_sis
@@ -47,7 +47,7 @@ equals(GENERATE_SIS_TARGETS, true) {
                               )
 
         ok_unsigned_sis_target.target = ok_unsigned_sis
-        ok_unsigned_sis_target.commands = createpackage.bat $(QT_SIS_OPTIONS) -o $$basename(TARGET)_template.pkg $(QT_SIS_TARGET)
+        ok_unsigned_sis_target.commands = createpackage.pl $(QT_SIS_OPTIONS) -o $$basename(TARGET)_template.pkg $(QT_SIS_TARGET)
 
         target_sis_target.target = $$basename(TARGET).sis
         target_sis_target.commands = $(MAKE) -f $(MAKEFILE) sis
@@ -61,7 +61,7 @@ equals(GENERATE_SIS_TARGETS, true) {
         installer_sis_target.depends = $$basename(TARGET).sis
 
         ok_installer_sis_target.target = ok_installer_sis
-        ok_installer_sis_target.commands = createpackage.bat $(QT_SIS_OPTIONS) $$basename(TARGET)_installer.pkg - \
+        ok_installer_sis_target.commands = createpackage.pl $(QT_SIS_OPTIONS) $$basename(TARGET)_installer.pkg - \
                                                $(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)
 
         fail_sis_nopkg_target.target = fail_sis_nopkg
@@ -86,7 +86,7 @@ equals(GENERATE_SIS_TARGETS, true) {
                                    )
 
         ok_stub_sis_target.target = ok_stub_sis
-        ok_stub_sis_target.commands = createpackage.bat -s $(QT_SIS_OPTIONS) $$basename(TARGET)_stub.pkg \
+        ok_stub_sis_target.commands = createpackage.pl -s $(QT_SIS_OPTIONS) $$basename(TARGET)_stub.pkg \
                                           $(QT_SIS_TARGET) $(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)
 
         QMAKE_EXTRA_TARGETS += sis_target \
@@ -118,7 +118,6 @@ equals(GENERATE_SIS_TARGETS, true) {
                 && echo $${shellFixedHash}  Version     :  >> $$make_cache_name \
                 && echo $${shellFixedHash} >> $$make_cache_name \
                 && echo $${shellFixedHash} ============================================================================== >> $$make_cache_name \
-                && echo. >> $$make_cache_name \
                 && echo QT_SIS_TARGET ?= $(QT_SIS_TARGET) >> $$make_cache_name
 
             QMAKE_EXTRA_TARGETS += store_build_target
