diff --git a/mkspecs/common/symbian/symbian.conf b/mkspecs/common/symbian/symbian.conf
index 7ec5c6c..d41aa00 100644
--- a/mkspecs/common/symbian/symbian.conf
+++ b/mkspecs/common/symbian/symbian.conf
@@ -95,7 +95,7 @@ contains(QMAKE_HOST.os,Windows) {
     QMAKE_COPY_DIR       = cp -r
     QMAKE_MOVE           = mv
     QMAKE_DEL_FILE       = rm -f
-    QMAKE_MKDIR          = mkdir
+    QMAKE_MKDIR          = mkdir -p
     QMAKE_DEL_DIR        = rmdir
     QMAKE_DEL_TREE       = rm -rf
     QMAKE_CHK_DIR_EXISTS = test -d
diff --git a/mkspecs/features/sis_targets.prf b/mkspecs/features/sis_targets.prf
index bcfdf8f..6566ec6 100644
--- a/mkspecs/features/sis_targets.prf
+++ b/mkspecs/features/sis_targets.prf
@@ -28,7 +28,7 @@ equals(GENERATE_SIS_TARGETS, true) {
                               )
 
         ok_sis_target.target = ok_sis
-        ok_sis_target.commands = createpackage.bat $(QT_SIS_OPTIONS) $$basename(TARGET)_template.pkg \
+        ok_sis_target.commands = createpackage $(QT_SIS_OPTIONS) $$basename(TARGET)_template.pkg \
                                      $(QT_SIS_TARGET) $(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)
 
         unsigned_sis_target.target = unsigned_sis
@@ -47,7 +47,7 @@ equals(GENERATE_SIS_TARGETS, true) {
                               )
 
         ok_unsigned_sis_target.target = ok_unsigned_sis
-        ok_unsigned_sis_target.commands = createpackage.bat $(QT_SIS_OPTIONS) -o $$basename(TARGET)_template.pkg $(QT_SIS_TARGET)
+        ok_unsigned_sis_target.commands = createpackage $(QT_SIS_OPTIONS) -o $$basename(TARGET)_template.pkg $(QT_SIS_TARGET)
 
         target_sis_target.target = $$basename(TARGET).sis
         target_sis_target.commands = $(MAKE) -f $(MAKEFILE) sis
@@ -61,7 +61,7 @@ equals(GENERATE_SIS_TARGETS, true) {
         installer_sis_target.depends = $$basename(TARGET).sis
 
         ok_installer_sis_target.target = ok_installer_sis
-        ok_installer_sis_target.commands = createpackage.bat $(QT_SIS_OPTIONS) $$basename(TARGET)_installer.pkg - \
+        ok_installer_sis_target.commands = createpackage $(QT_SIS_OPTIONS) $$basename(TARGET)_installer.pkg - \
                                                $(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)
 
         fail_sis_nopkg_target.target = fail_sis_nopkg
@@ -86,7 +86,7 @@ equals(GENERATE_SIS_TARGETS, true) {
                                    )
 
         ok_stub_sis_target.target = ok_stub_sis
-        ok_stub_sis_target.commands = createpackage.bat -s $(QT_SIS_OPTIONS) $$basename(TARGET)_stub.pkg \
+        ok_stub_sis_target.commands = createpackage -s $(QT_SIS_OPTIONS) $$basename(TARGET)_stub.pkg \
                                           $(QT_SIS_TARGET) $(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)
 
         QMAKE_EXTRA_TARGETS += sis_target \
@@ -118,7 +118,6 @@ equals(GENERATE_SIS_TARGETS, true) {
                 && echo $${shellFixedHash}  Version     :  >> $$make_cache_name \
                 && echo $${shellFixedHash} >> $$make_cache_name \
                 && echo $${shellFixedHash} ============================================================================== >> $$make_cache_name \
-                && echo. >> $$make_cache_name \
                 && echo QT_SIS_TARGET ?= $(QT_SIS_TARGET) >> $$make_cache_name
 
             QMAKE_EXTRA_TARGETS += store_build_target
diff --git a/mkspecs/features/symbian/symbian_building.prf b/mkspecs/features/symbian/symbian_building.prf
index ce124ec..13748a2 100644
--- a/mkspecs/features/symbian/symbian_building.prf
+++ b/mkspecs/features/symbian/symbian_building.prf
@@ -221,7 +221,7 @@ symbian_resources_INCLUDES = $$replace(symbian_resources_INCLUDES, ",", " -I")
 symbian_resources_INCLUDES += $$join(INCLUDEPATH, " -I", "-I")
 symbian_resources_DEFINES = $$join(DEFINES, " -D", "-D")
 symbian_resources_RCC_DIR = $$replace(RCC_DIR, "/$", "")
-symbian_resources_INCLUDES += "-I $$symbian_resources_RCC_DIR"
+symbian_resources_INCLUDES += "-I$$symbian_resources_RCC_DIR"
 
 for(symbian_resource, SYMBIAN_RESOURCES) {
     symbian_resource = $$basename(symbian_resource)
@@ -237,7 +237,7 @@ symbianresources.commands = cpp -nostdinc -undef \
                             $$symbian_resources_INCLUDES \
                             $$symbian_resources_DEFINES \
                             ${QMAKE_FILE_NAME} \
-                            -o $${symbian_resources_RCC_DIR}/${QMAKE_FILE_BASE}.rpp \
+                            > $${symbian_resources_RCC_DIR}/${QMAKE_FILE_BASE}.rpp \
                             && rcomp -u -m045,046,047 \
                             -s$${symbian_resources_RCC_DIR}/${QMAKE_FILE_BASE}.rpp \
                             -o$${symbianDestdir}/${QMAKE_FILE_BASE}$${QT_LIBINFIX}.rsc \
@@ -260,7 +260,7 @@ contains(TEMPLATE, "app"):!contains(CONFIG, "no_icon") {
                                   $$symbian_resources_INCLUDES \
                                   $$symbian_resources_DEFINES \
                                   $${baseResourceTarget}.rss \
-                                  -o $${symbian_resources_RCC_DIR}/$${baseResourceTarget}.rpp \
+                                  > $${symbian_resources_RCC_DIR}/$${baseResourceTarget}.rpp \
                                   && rcomp -u -m045,046,047 \
                                   -s$${symbian_resources_RCC_DIR}/$${baseResourceTarget}.rpp \
                                   -o$${symbianDestdir}/$${baseResourceTarget}.rsc \
@@ -278,7 +278,7 @@ contains(TEMPLATE, "app"):!contains(CONFIG, "no_icon") {
                                      $$symbian_resources_INCLUDES \
                                      $$symbian_resources_DEFINES \
                                      $${baseResourceTarget}_reg.rss \
-                                     -o $${symbian_resources_RCC_DIR}/$${baseResourceTarget}_reg.rpp \
+                                     > $${symbian_resources_RCC_DIR}/$${baseResourceTarget}_reg.rpp \
                                      && rcomp -u -m045,046,047 \
                                      -s$${symbian_resources_RCC_DIR}/$${baseResourceTarget}_reg.rpp \
                                      -o$${symbianDestdir}/$${baseResourceTarget}_reg.rsc \
diff --git a/mkspecs/symbian/linux-gcce/qmake.conf b/mkspecs/symbian/linux-gcce/qmake.conf
index faac2f1..09852ca 100644
--- a/mkspecs/symbian/linux-gcce/qmake.conf
+++ b/mkspecs/symbian/linux-gcce/qmake.conf
@@ -54,7 +54,7 @@ DEFINES		+= __GCCE__ \
                    _STLP_NO_EXCEPTION_HEADER
 
 QMAKE_LFLAGS_APP        += --entry=_E32Startup -u _E32Startup
-QMAKE_LFLAGS_SHLIB      += --default-symver --entry _E32Dll
+QMAKE_LFLAGS_SHLIB      += --default-symver --entry=_E32Dll -u _E32Dll
 QMAKE_LFLAGS_PLUGIN     += $$QMAKE_LFLAGS_SHLIB
 
 gcceExtraFlags = --include=${EPOCROOT}/epoc32/include/gcce/gcce.h -march=armv5t -mapcs -mthumb-interwork -nostdinc -c -msoft-float -T script
@@ -90,6 +90,5 @@ INCLUDEPATH = ${EPOCROOT}/epoc32/include/ \
     $${EPOCROOT}/epoc32/include/stdapis \
     $${EPOCROOT}/epoc32/include/gcce \
     ${EPOCROOT}/epoc32/include/stdapis/sys \
-    ${EPOCROOT}/epoc32/include/stdapis/stlport \
     $$INCLUDEPATH
 
diff --git a/qmake/generators/symbian/symbiancommon.cpp b/qmake/generators/symbian/symbiancommon.cpp
index 58d2edd..a3606bd 100644
--- a/qmake/generators/symbian/symbiancommon.cpp
+++ b/qmake/generators/symbian/symbiancommon.cpp
@@ -390,9 +390,9 @@ void SymbianCommonGenerator::generatePkgFile(const QString &iconFile, bool epocB
                     t << QString("\"%1epoc32/data/z%2\" - \"!:%3\"")
                          .arg(epocRoot())
                          .arg(iconFile)
-                         .arg(QDir::toNativeSeparators(iconFile)) << endl << endl;
+                         .arg(QString(iconFile).replace("/", "\\")) << endl << endl;
                     ts << QString("\"\" - \"%1\"")
-                         .arg(romPath(QDir::toNativeSeparators(iconFile))) << endl << endl;
+                         .arg(romPath(QString(iconFile).replace("/", "\\"))) << endl << endl;
                 } else {
                     QDir mifIconDir(project->first("DESTDIR"));
                     QFileInfo mifIcon(mifIconDir.relativeFilePath(project->first("TARGET")));
@@ -401,9 +401,9 @@ void SymbianCommonGenerator::generatePkgFile(const QString &iconFile, bool epocB
                     t << QString("\"%1/%2\" - \"!:%3\"")
                          .arg(mifIcon.path())
                          .arg(mifIconFileName)
-                         .arg(QDir::toNativeSeparators(iconFile)) << endl << endl;
+                         .arg(QString(iconFile).replace("/", "\\")) << endl << endl;
                     ts << QString("\"\" - \"%1\"")
-                         .arg(romPath(QDir::toNativeSeparators(iconFile))) << endl << endl;
+                         .arg(romPath(QString(iconFile).replace("/", "\\"))) << endl << endl;
                 }
             }
         }
diff --git a/qmake/generators/symbian/symmake_abld.cpp b/qmake/generators/symbian/symmake_abld.cpp
index 03d1a18..4fbe6f7 100644
--- a/qmake/generators/symbian/symmake_abld.cpp
+++ b/qmake/generators/symbian/symmake_abld.cpp
@@ -209,8 +209,11 @@ void SymbianAbldMakefileGenerator::writeWrapperMakefile(QFile& wrapperFile, bool
 #ifdef Q_OS_WIN32
     t << "XCOPY             = xcopy /d /f /h /r /y /i" << endl;
     t << "ABLD              = ABLD.BAT" << endl;
+#elif defined(Q_OS_MAC)
+    t << "XCOPY             = cp -R -v" << endl;
+    t << "ABLD              = abld" << endl;
 #else
-    t << "XCOPY             = cp -u -v" << endl;
+    t << "XCOPY             = cp -R -u -v" << endl;
     t << "ABLD              = abld" << endl;
 #endif
     t << "DEBUG_PLATFORMS   = " << debugPlatforms.join(" ") << endl;
@@ -327,6 +330,7 @@ void SymbianAbldMakefileGenerator::writeWrapperMakefile(QFile& wrapperFile, bool
                     QString fixedValue(QDir::toNativeSeparators(values.at(i)));
                     dirsToClean << fixedValue;
                     t << "\t-@ " << dirExists << " \""  << fixedValue << "\" "
+                      << (isWindowsShell() ? "" : "|| ")
                       << mkdir << " \"" << fixedValue << "\"" << endl;
                 }
             }
@@ -445,8 +449,14 @@ bool SymbianAbldMakefileGenerator::writeDeploymentTargets(QTextStream &t, bool i
         // Xcopy prompts for selecting file or directory if target doesn't exist,
         // and doesn't provide switch to force file selection. It does provide dir forcing, though,
         // so strip the last part of the destination.
+        QString dir = Option::fixPathToLocalOS(depList.at(i).to.left(depList.at(i).to.lastIndexOf("\\") + 1));
+        QString dirExists = var("QMAKE_CHK_DIR_EXISTS");
+        QString mkdir = var("QMAKE_MKDIR");
+        t << "\t-@ " << dirExists << " \""  << dir << "\" "
+                      << (isWindowsShell() ? "" : "|| ")
+                      << mkdir << " \"" << dir << "\"" << endl;
         t << "\t-$(XCOPY) \"" << depList.at(i).from << "\" \""
-          << depList.at(i).to.left(depList.at(i).to.lastIndexOf("\\") + 1) << "\"" << endl;
+          << dir << "\"" << endl;
     }
 
     t << endl;
