diff --git a/mkspecs/symbian/linux-gcce/qmake.conf b/mkspecs/symbian/linux-gcce/qmake.conf
index faac2f1..09852ca 100644
--- a/mkspecs/symbian/linux-gcce/qmake.conf
+++ b/mkspecs/symbian/linux-gcce/qmake.conf
@@ -54,7 +54,7 @@ DEFINES		+= __GCCE__ \
                    _STLP_NO_EXCEPTION_HEADER
 
 QMAKE_LFLAGS_APP        += --entry=_E32Startup -u _E32Startup
-QMAKE_LFLAGS_SHLIB      += --default-symver --entry _E32Dll
+QMAKE_LFLAGS_SHLIB      += --default-symver --entry=_E32Dll -u _E32Dll
 QMAKE_LFLAGS_PLUGIN     += $$QMAKE_LFLAGS_SHLIB
 
 gcceExtraFlags = --include=${EPOCROOT}/epoc32/include/gcce/gcce.h -march=armv5t -mapcs -mthumb-interwork -nostdinc -c -msoft-float -T script
@@ -90,6 +90,5 @@ INCLUDEPATH = ${EPOCROOT}/epoc32/include/ \
     $${EPOCROOT}/epoc32/include/stdapis \
     $${EPOCROOT}/epoc32/include/gcce \
     ${EPOCROOT}/epoc32/include/stdapis/sys \
-    ${EPOCROOT}/epoc32/include/stdapis/stlport \
     $$INCLUDEPATH
 
