diff --git a/bin/createpackage.pl b/bin/createpackage.pl
index 7f4375f..65db28d 100755
--- a/bin/createpackage.pl
+++ b/bin/createpackage.pl
@@ -163,9 +163,9 @@ my $stub_sis_name = $sisoutputbasename."_stub.sis";
 my $scriptpath = dirname(__FILE__);
 my $certtext = $certificate;
 my $certpath = $scriptpath;
-$certpath =~ s-^(.*[^\\])$-$1\\-o;          # ensure path ends with a backslash
-$certpath =~ s-/-\\-go;                     # for those working with UNIX shells
-$certpath =~ s-bin\\$-src\\s60installs\\-;  # certificates are one step up in hierarcy
+$certpath =~ s-^(.*[^\/])$-$1\/-o;          # ensure path ends with a backslash
+$certpath =~ s-\\-/-go;                     # for those working with UNIX shells
+$certpath =~ s-bin/$-src/s60installs/-;     # certificates are one step up in hierarcy
 
 # Check some pre-conditions and print error messages if needed.
 unless (length($templatepkg)) {
diff --git a/mkspecs/common/symbian/symbian.conf b/mkspecs/common/symbian/symbian.conf
index ba31116..7e3784c 100644
--- a/mkspecs/common/symbian/symbian.conf
+++ b/mkspecs/common/symbian/symbian.conf
@@ -95,14 +95,14 @@ contains(QMAKE_HOST.os,Windows) {
     QMAKE_COPY_DIR       = cp -r
     QMAKE_MOVE           = mv
     QMAKE_DEL_FILE       = rm -f
-    QMAKE_MKDIR          = mkdir
+    QMAKE_MKDIR          = mkdir -p
     QMAKE_DEL_DIR        = rmdir
     QMAKE_CHK_DIR_EXISTS = test -d
 }
 
-QMAKE_MOC		= $$[QT_INSTALL_BINS]$${DIR_SEPARATOR}moc.exe
-QMAKE_UIC		= $$[QT_INSTALL_BINS]$${DIR_SEPARATOR}uic.exe
-QMAKE_IDC		= $$[QT_INSTALL_BINS]$${DIR_SEPARATOR}idc.exe
+QMAKE_MOC		= moc
+QMAKE_UIC		= uic
+QMAKE_IDC		= idc
 
 QMAKE_IDL		= midl
 QMAKE_LIB		= ar -ru
diff --git a/qmake/generators/makefile.cpp b/qmake/generators/makefile.cpp
index ad865a2..527bd0d 100644
--- a/qmake/generators/makefile.cpp
+++ b/qmake/generators/makefile.cpp
@@ -1822,7 +1822,7 @@ MakefileGenerator::writeExtraCompilerTargets(QTextStream &t)
             if(tmp_clean.indexOf("${QMAKE_") == -1) {
                 t << "\n\t" << "-$(DEL_FILE) " << tmp_clean;
                 if (isForSymbian())
-                    t << " 2> NUL"; // Eliminate unnecessary warnings
+                    t << " 2> /dev/null"; // Eliminate unnecessary warnings
                 wrote_clean = true;
             }
             if(!wrote_clean_cmds || !wrote_clean) {
@@ -1852,7 +1852,7 @@ MakefileGenerator::writeExtraCompilerTargets(QTextStream &t)
                 }
                 if(!cleans.isEmpty()) {
                     if (isForSymbian())
-                        t << valGlue(cleans, "\n\t" + del_statement, " 2> NUL\n\t" + del_statement, " 2> NUL");
+                        t << valGlue(cleans, "\n\t" + del_statement, " 2> /dev/null\n\t" + del_statement, " 2> /dev/null");
                     else
                         t << valGlue(cleans, "\n\t" + del_statement, "\n\t" + del_statement, "");
                 }
diff --git a/qmake/generators/symbian/symmake.cpp b/qmake/generators/symbian/symmake.cpp
index 84f0be4..8ac810a 100644
--- a/qmake/generators/symbian/symmake.cpp
+++ b/qmake/generators/symbian/symmake.cpp
@@ -468,7 +468,7 @@ void SymbianMakefileGenerator::generatePkgFile(const QString &iconFile, Deployme
                 t << QString("\"%1epoc32/data/z%2\"    - \"!:%3\"")
                      .arg(epocRoot())
                      .arg(iconFile)
-                     .arg(QDir::toNativeSeparators(iconFile)) << endl << endl;
+                     .arg(QString(iconFile).replace("/", "\\")) << endl << endl;
             }
         }
     }
@@ -1942,7 +1942,7 @@ void SymbianMakefileGenerator::writeSisTargets(QTextStream &t)
 
     t << OK_SIS_TARGET ":" << endl;
 
-    QString pkgcommand = QString::fromLatin1("\tcreatepackage.bat $(QT_SIS_OPTIONS) %1_template.%2 $(QT_SIS_TARGET) " \
+    QString pkgcommand = QString::fromLatin1("\tcreatepackage.pl $(QT_SIS_OPTIONS) %1_template.%2 $(QT_SIS_TARGET) " \
                                  "$(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)")
                           .arg(fixedTarget)
                           .arg("pkg");
@@ -1992,7 +1992,7 @@ void SymbianMakefileGenerator::writeSisTargets(QTextStream &t)
 
     t << OK_INSTALLER_SIS_TARGET ": " << endl;
 
-    pkgcommand = QString::fromLatin1("\tcreatepackage.bat $(QT_SIS_OPTIONS) %1_installer.%2 - " \
+    pkgcommand = QString::fromLatin1("\tcreatepackage.pl $(QT_SIS_OPTIONS) %1_installer.%2 - " \
                          "$(QT_SIS_CERTIFICATE) $(QT_SIS_KEY) $(QT_SIS_PASSPHRASE)")
                   .arg(fixedTarget)
                   .arg("pkg");
diff --git a/qmake/generators/symbian/symmake_abld.cpp b/qmake/generators/symbian/symmake_abld.cpp
index 70264a5..c1c7c95 100644
--- a/qmake/generators/symbian/symmake_abld.cpp
+++ b/qmake/generators/symbian/symmake_abld.cpp
@@ -206,16 +206,17 @@ void SymbianAbldMakefileGenerator::writeWrapperMakefile(QFile& wrapperFile, bool
     t << "MOVE              = " << var("QMAKE_MOVE") << endl;
     t << "CHK_DIR_EXISTS    = " << var("QMAKE_CHK_DIR_EXISTS") << endl;
     t << "MKDIR             = " << var("QMAKE_MKDIR") << endl;
-    t << "XCOPY             = xcopy /d /f /h /r /y /i" << endl;
-    t << "ABLD              = ABLD.BAT" << endl;
+    t << "XCOPY             = cp -R -v" << endl;
+    t << "ABLD              = abld" << endl;
     t << "DEBUG_PLATFORMS   = " << debugPlatforms.join(" ") << endl;
     t << "RELEASE_PLATFORMS = " << releasePlatforms.join(" ") << endl;
     t << "MAKE              = make" << endl;
     t << endl;
+    t << "EPOCROOT:=$(subst \\,/,$(EPOCROOT))" << endl;
     t << "ifeq (WINS,$(findstring WINS, $(PLATFORM)))" << endl;
-    t << "ZDIR=$(EPOCROOT)epoc32\\release\\$(PLATFORM)\\$(CFG)\\Z" << endl;
+    t << "ZDIR=$(EPOCROOT)epoc32/release/$(PLATFORM)/$(CFG)/z" << endl;
     t << "else" << endl;
-    t << "ZDIR=$(EPOCROOT)epoc32\\data\\z" << endl;
+    t << "ZDIR=$(EPOCROOT)epoc32/data/z" << endl;
     t << "endif" << endl;
     t << endl;
     t << "DEFINES" << '\t' << " = "
@@ -315,8 +316,8 @@ void SymbianAbldMakefileGenerator::writeWrapperMakefile(QFile& wrapperFile, bool
                 if (values.at(i).endsWith("/" QT_EXTRA_INCLUDE_DIR)) {
                     QString fixedValue(QDir::toNativeSeparators(values.at(i)));
                     dirsToClean << fixedValue;
-                    t << "\t-@ if NOT EXIST \""  << fixedValue << "\" mkdir \""
-                      << fixedValue << "\"" << endl;
+                    t << "\t-@ if [ ! -d \""  << fixedValue << "\" ]; then mkdir \""
+                      << fixedValue << "\"; fi" << endl;
                 }
             }
         }
@@ -440,8 +441,14 @@ bool SymbianAbldMakefileGenerator::writeDeploymentTargets(QTextStream &t, bool i
         // Xcopy prompts for selecting file or directory if target doesn't exist,
         // and doesn't provide switch to force file selection. It does provide dir forcing, though,
         // so strip the last part of the destination.
+        QString dir = Option::fixPathToLocalOS(depList.at(i).to.left(depList.at(i).to.lastIndexOf("\\") + 1));
+        QString dirExists = var("QMAKE_CHK_DIR_EXISTS");
+        QString mkdir = var("QMAKE_MKDIR");
+        t << "\t-@ " << dirExists << " \""  << dir << "\" "
+                      << (isWindowsShell() ? "" : "|| ")
+                      << mkdir << " \"" << dir << "\"" << endl;
         t << "\t-$(XCOPY) \"" << depList.at(i).from << "\" \""
-          << depList.at(i).to.left(depList.at(i).to.lastIndexOf("\\") + 1) << "\"" << endl;
+          << dir << "\"" << endl;
     }
 
     t << endl;
@@ -468,16 +475,15 @@ bool SymbianAbldMakefileGenerator::writeDeploymentTargets(QTextStream &t, bool i
 void SymbianAbldMakefileGenerator::writeStoreBuildTarget(QTextStream &t)
 {
     t << STORE_BUILD_TARGET ":" << endl;
-    t << "\t@echo # ============================================================================== > " MAKE_CACHE_NAME << endl;
-    t << "\t@echo # This file is generated by make and should not be modified by the user >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo #  Name        : " << MAKE_CACHE_NAME << " >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo #  Part of     : " << project->values("TARGET").join(" ") << " >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo #  Description : This file is used to cache last build target for >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo #                make sis target. >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo #  Version     :  >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo # >> " MAKE_CACHE_NAME << endl;
-    t << "\t@echo # ============================================================================== >> " MAKE_CACHE_NAME <<  endl;
-    t << "\t@echo. >> " MAKE_CACHE_NAME <<  endl;
+    t << "\t@echo \\# ============================================================================== > " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\# This file is generated by make and should not be modified by the user >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\#  Name        : " << MAKE_CACHE_NAME << " >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\#  Part of     : " << project->values("TARGET").join(" ") << " >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\#  Description : This file is used to cache last build target for >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\#                make sis target. >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\#  Version     :  >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\# >> " MAKE_CACHE_NAME << endl;
+    t << "\t@echo \\# ============================================================================== >> " MAKE_CACHE_NAME <<  endl;
     t << "\t@echo QT_SIS_TARGET ?= $(QT_SIS_TARGET) >> " MAKE_CACHE_NAME << endl;
     t << endl;
 
diff --git a/qmake/project.cpp b/qmake/project.cpp
index 5a87dd7..17e561a 100644
--- a/qmake/project.cpp
+++ b/qmake/project.cpp
@@ -593,9 +593,13 @@ QStringList qmake_feature_paths(QMakeProperty *prop=0)
         const QString base_concat = QDir::separator() + QString("features");
         switch(Option::target_mode) {
         case Option::TARG_MACX_MODE:                     //also a unix
-            concat << base_concat + QDir::separator() + "mac";
-            concat << base_concat + QDir::separator() + "macx";
-            concat << base_concat + QDir::separator() + "unix";
+            if (isForSymbian())
+                concat << base_concat + QDir::separator() + "symbian";
+            else {
+                concat << base_concat + QDir::separator() + "mac";
+                concat << base_concat + QDir::separator() + "macx";
+                concat << base_concat + QDir::separator() + "unix";
+            }
             break;
         case Option::TARG_UNIX_MODE:
             {
diff --git a/src/gui/s60framework/qs60maindocument.h b/src/gui/s60framework/qs60maindocument.h
index 79eadd7..90d69a4 100644
--- a/src/gui/s60framework/qs60maindocument.h
+++ b/src/gui/s60framework/qs60maindocument.h
@@ -46,7 +46,7 @@
 
 #ifdef Q_WS_S60
 
-#include <AknDoc.h>
+#include <akndoc.h>
 
 class CEikApplication;
 
