--- createpackage.pl.orig	2010-08-08 11:57:01.000000000 +0300
+++ createpackage.pl	2010-08-08 11:55:37.000000000 +0300
@@ -163,9 +163,9 @@
 my $scriptpath = dirname(__FILE__);
 my $certtext = $certificate;
 my $certpath = $scriptpath;
-$certpath =~ s-^(.*[^\\])$-$1\\-o;          # ensure path ends with a backslash
-$certpath =~ s-/-\\-go;                     # for those working with UNIX shells
-$certpath =~ s-bin\\$-src\\s60installs\\-;  # certificates are one step up in hierarcy
+$certpath =~ s-^(.*[^\/])$-$1\/-o;          # ensure path ends with a backslash
+$certpath =~ s-\\-/-go;                     # for those working with UNIX shells
+$certpath =~ s-bin/$-src/s60installs/-;     # certificates are one step up in hierarcy
 
 # Check some pre-conditions and print error messages if needed.
 unless (length($templatepkg)) {
