diff -urN src-orig/bfd/archive.c src/bfd/archive.c
--- src-orig/bfd/archive.c	2002-10-22 13:13:54.000000000 +0300
+++ src/bfd/archive.c	2014-07-31 10:29:34.000000000 +0300
@@ -167,6 +167,22 @@
 #define arch_eltdata(bfd) ((struct areltdata *)((bfd)->arelt_data))
 #define arch_hdr(bfd) ((struct ar_hdr *)arch_eltdata(bfd)->arch_header)
 
+void
+_bfd_ar_spacepad (char *p, size_t n, const char *fmt, long val)
+{
+  static char buf[20];
+  size_t len;
+  snprintf (buf, sizeof (buf), fmt, val);
+  len = strlen (buf);
+  if (len < n)
+    {
+      memcpy (p, buf, len);
+      memset (p + len, ' ', n - len);
+    }
+  else
+    memcpy (p, buf, n);
+}
+
 static char *get_extended_arelt_filename PARAMS ((bfd *arch,
 						  const char *name));
 static boolean do_slurp_bsd_armap PARAMS ((bfd *abfd));
@@ -1317,17 +1333,8 @@
 	      strptr[thislen + 1] = '\012';
 	    }
 	  hdr->ar_name[0] = ar_padchar (current);
-	  /* We know there will always be enough room (one of the few
-	     cases where you may safely use sprintf). */
-	  sprintf ((hdr->ar_name) + 1, "%-d", (unsigned) (strptr - *tabloc));
-	  /* Kinda Kludgy.  We should just use the returned value of
-	     sprintf but not all implementations get this right */
-	  {
-	    char *temp = hdr->ar_name + 2;
-	    for (; temp < hdr->ar_name + maxname; temp++)
-	      if (*temp == '\0')
-		*temp = ' ';
-	  }
+          _bfd_ar_spacepad (hdr->ar_name + 1, maxname - 1, "%-ld",
+                            strptr - *tabloc);
 	  strptr += thislen + 1;
 	  if (trailing_slash)
 	    ++strptr;
@@ -1368,25 +1375,17 @@
   /* ar headers are space padded, not null padded! */
   memset ((PTR) hdr, ' ', sizeof (struct ar_hdr));
 
-  strncpy (hdr->ar_fmag, ARFMAG, 2);
-
-  /* Goddamned sprintf doesn't permit MAXIMUM field lengths */
-  sprintf ((hdr->ar_date), "%-12ld", (long) status.st_mtime);
-  sprintf ((hdr->ar_uid), "%ld", (long) status.st_uid);
-  sprintf ((hdr->ar_gid), "%ld", (long) status.st_gid);
-  sprintf ((hdr->ar_mode), "%-8o", (unsigned int) status.st_mode);
-  sprintf ((hdr->ar_size), "%-10ld", (long) status.st_size);
-  /* Correct for a lossage in sprintf whereby it null-terminates.  I cannot
-     understand how these C losers could design such a ramshackle bunch of
-     IO operations */
-  temp = (char *) hdr;
-  temp1 = temp + sizeof (struct ar_hdr) - 2;
-  for (; temp < temp1; temp++)
-    {
-      if (*temp == '\0')
-	*temp = ' ';
-    }
-  strncpy (hdr->ar_fmag, ARFMAG, 2);
+  _bfd_ar_spacepad (hdr->ar_date, sizeof (hdr->ar_date), "%-12ld",
+                    status.st_mtime);
+  _bfd_ar_spacepad (hdr->ar_uid, sizeof (hdr->ar_uid), "%ld",
+                    status.st_uid);
+  _bfd_ar_spacepad (hdr->ar_gid, sizeof (hdr->ar_gid), "%ld",
+                    status.st_gid);
+  _bfd_ar_spacepad (hdr->ar_mode, sizeof (hdr->ar_mode), "%-8lo",
+                    status.st_mode);
+  _bfd_ar_spacepad (hdr->ar_size, sizeof (hdr->ar_size), "%-10ld",
+                    status.st_size);
+  memcpy (hdr->ar_fmag, ARFMAG, 2);
   ared->parsed_size = status.st_size;
   ared->arch_header = (char *) hdr;
 
@@ -1632,15 +1631,12 @@
     {
       struct ar_hdr hdr;
 
-      memset ((char *) (&hdr), 0, sizeof (struct ar_hdr));
-      strcpy (hdr.ar_name, ename);
+      memset (&hdr, ' ', sizeof (struct ar_hdr));
+      memcpy (hdr.ar_name, ename, strlen (ename));
       /* Round size up to even number in archive header.  */
-      sprintf (&(hdr.ar_size[0]), "%-10d",
-	       (int) ((elength + 1) & ~1));
-      strncpy (hdr.ar_fmag, ARFMAG, 2);
-      for (i = 0; i < sizeof (struct ar_hdr); i++)
-	if (((char *) (&hdr))[i] == '\0')
-	  (((char *) (&hdr))[i]) = ' ';
+      _bfd_ar_spacepad (hdr.ar_size, sizeof (hdr.ar_size), "%-10ld",
+                        (elength + 1) & ~(bfd_size_type) 1);
+      memcpy (hdr.ar_fmag, ARFMAG, 2);
       if ((bfd_write ((char *) &hdr, 1, sizeof (struct ar_hdr), arch)
 	   != sizeof (struct ar_hdr))
 	  || bfd_write (etable, 1, elength, arch) != elength)
@@ -1876,30 +1872,27 @@
   unsigned int count;
   struct ar_hdr hdr;
   struct stat statbuf;
-  unsigned int i;
 
   firstreal = mapsize + elength + sizeof (struct ar_hdr) + SARMAG;
 
   stat (arch->filename, &statbuf);
-  memset ((char *) (&hdr), 0, sizeof (struct ar_hdr));
-  sprintf (hdr.ar_name, RANLIBMAG);
+  memset (&hdr, ' ', sizeof (struct ar_hdr));
+  memcpy (hdr.ar_name, RANLIBMAG, strlen (RANLIBMAG));
   /* Remember the timestamp, to keep it holy.  But fudge it a little.  */
   bfd_ardata (arch)->armap_timestamp = statbuf.st_mtime + ARMAP_TIME_OFFSET;
   bfd_ardata (arch)->armap_datepos = (SARMAG
 				      + offsetof (struct ar_hdr, ar_date[0]));
-  sprintf (hdr.ar_date, "%ld", bfd_ardata (arch)->armap_timestamp);
+  _bfd_ar_spacepad (hdr.ar_date, sizeof (hdr.ar_date), "%ld",
+                    bfd_ardata (arch)->armap_timestamp);
 #ifndef _WIN32
-  sprintf (hdr.ar_uid, "%ld", (long) getuid ());
-  sprintf (hdr.ar_gid, "%ld", (long) getgid ());
+  _bfd_ar_spacepad (hdr.ar_uid, sizeof (hdr.ar_uid), "%ld", getuid ());
+  _bfd_ar_spacepad (hdr.ar_gid, sizeof (hdr.ar_gid), "%ld", getgid ());
 #else
-  sprintf (hdr.ar_uid, "%ld", (long) 666);
-  sprintf (hdr.ar_gid, "%ld", (long) 42);
+  _bfd_ar_spacepad (hdr.ar_uid, sizeof (hdr.ar_uid), "%ld", 666);
+  _bfd_ar_spacepad (hdr.ar_gid, sizeof (hdr.ar_gid), "%ld", 42);
 #endif
-  sprintf (hdr.ar_size, "%-10d", (int) mapsize);
-  strncpy (hdr.ar_fmag, ARFMAG, 2);
-  for (i = 0; i < sizeof (struct ar_hdr); i++)
-    if (((char *) (&hdr))[i] == '\0')
-      (((char *) (&hdr))[i]) = ' ';
+  _bfd_ar_spacepad (hdr.ar_size, sizeof (hdr.ar_size), "%-10ld", mapsize);
+  memcpy (hdr.ar_fmag, ARFMAG, 2);
   if (bfd_write ((char *) &hdr, 1, sizeof (struct ar_hdr), arch)
       != sizeof (struct ar_hdr))
     return false;
@@ -1964,7 +1957,6 @@
 {
   struct stat archstat;
   struct ar_hdr hdr;
-  unsigned int i;
 
   /* Flush writes, get last-write timestamp from file, and compare it
      to the timestamp IN the file.  */
@@ -1981,11 +1973,9 @@
   bfd_ardata (arch)->armap_timestamp = archstat.st_mtime + ARMAP_TIME_OFFSET;
 
   /* Prepare an ASCII version suitable for writing.  */
-  memset (hdr.ar_date, 0, sizeof (hdr.ar_date));
-  sprintf (hdr.ar_date, "%ld", bfd_ardata (arch)->armap_timestamp);
-  for (i = 0; i < sizeof (hdr.ar_date); i++)
-    if (hdr.ar_date[i] == '\0')
-      (hdr.ar_date)[i] = ' ';
+  memset (hdr.ar_date, ' ', sizeof (hdr.ar_date));
+  _bfd_ar_spacepad (hdr.ar_date, sizeof (hdr.ar_date), "%ld",
+                    bfd_ardata (arch)->armap_timestamp);
 
   /* Write it into the file.  */
   bfd_ardata (arch)->armap_datepos = (SARMAG
@@ -2033,7 +2023,6 @@
   bfd *current = arch->archive_head;
   unsigned int count;
   struct ar_hdr hdr;
-  unsigned int i;
   int padit = mapsize & 1;
 
   if (padit)
@@ -2045,19 +2034,17 @@
 			     + sizeof (struct ar_hdr)
 			     + SARMAG);
 
-  memset ((char *) (&hdr), 0, sizeof (struct ar_hdr));
+  memset (&hdr, ' ', sizeof (struct ar_hdr));
   hdr.ar_name[0] = '/';
-  sprintf (hdr.ar_size, "%-10d", (int) mapsize);
-  sprintf (hdr.ar_date, "%ld", (long) time (NULL));
+  _bfd_ar_spacepad (hdr.ar_size, sizeof (hdr.ar_size), "%-10ld",
+                    mapsize);
+  _bfd_ar_spacepad (hdr.ar_date, sizeof (hdr.ar_date), "%ld",
+                    time (NULL));
   /* This, at least, is what Intel coff sets the values to.: */
-  sprintf ((hdr.ar_uid), "%d", 0);
-  sprintf ((hdr.ar_gid), "%d", 0);
-  sprintf ((hdr.ar_mode), "%-7o", (unsigned) 0);
-  strncpy (hdr.ar_fmag, ARFMAG, 2);
-
-  for (i = 0; i < sizeof (struct ar_hdr); i++)
-    if (((char *) (&hdr))[i] == '\0')
-      (((char *) (&hdr))[i]) = ' ';
+  _bfd_ar_spacepad (hdr.ar_uid, sizeof (hdr.ar_uid), "%ld", 0);
+  _bfd_ar_spacepad (hdr.ar_gid, sizeof (hdr.ar_gid), "%ld", 0);
+  _bfd_ar_spacepad (hdr.ar_mode, sizeof (hdr.ar_mode), "%-7lo", 0);
+  memcpy (hdr.ar_fmag, ARFMAG, 2);
 
   /* Write the ar header for this item and the number of symbols */
 
diff -urN src-orig/bfd/bfd-in2.h src/bfd/bfd-in2.h
--- src-orig/bfd/bfd-in2.h	2002-10-22 13:13:54.000000000 +0300
+++ src/bfd/bfd-in2.h	2014-07-31 10:29:34.000000000 +0300
@@ -701,20 +701,26 @@
 /* ARM Interworking support.  Called from linker.  */
 #ifdef COFF_IMAGE_WITH_PE
 static
+#else 
+extern
 #endif
-extern boolean bfd_arm_allocate_interworking_sections
+boolean bfd_arm_allocate_interworking_sections
   PARAMS ((struct bfd_link_info *));
 
 #ifdef COFF_IMAGE_WITH_PE
 static
+#else 
+extern
 #endif
-extern boolean bfd_arm_process_before_allocation
+boolean bfd_arm_process_before_allocation
   PARAMS ((bfd *, struct bfd_link_info *, int));
 
 #ifdef COFF_IMAGE_WITH_PE
 static
+#else 
+extern
 #endif
-extern boolean bfd_arm_get_bfd_for_interworking
+boolean bfd_arm_get_bfd_for_interworking
   PARAMS ((bfd *, struct bfd_link_info *));
 
 /* And more from the source.  */
diff -urN src-orig/bfd/pei-arm.c src/bfd/pei-arm.c
--- src-orig/bfd/pei-arm.c	2002-10-22 13:13:55.000000000 +0300
+++ src/bfd/pei-arm.c	2014-07-31 10:29:34.000000000 +0300
@@ -17,7 +17,6 @@
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
 
-#include "bfd.h"
 #include "sysdep.h"
 
 #ifndef TARGET_LITTLE_SYM
@@ -33,4 +32,5 @@
 #define PCRELOFFSET         true
 #define COFF_LONG_SECTION_NAMES
 
+#include "bfd.h"
 #include "coff-arm.c"
diff -urN src-orig/config.guess src/config.guess
--- src-orig/config.guess	2002-10-22 13:13:56.000000000 +0300
+++ src/config.guess	2014-07-31 10:29:34.000000000 +0300
@@ -219,6 +219,15 @@
     mvme88k:OpenBSD:*:*)
 	echo m88k-unknown-openbsd${UNAME_RELEASE}
 	exit 0 ;;
+    Power*Macintosh:Darwin:*:*)
+	echo powerpc-apple-darwin${UNAME_RELEASE}
+	exit 0 ;;
+    i386:Darwin:*:*)
+	echo i386-apple-darwin${UNAME_RELEASE}
+	exit 0 ;;
+    x86_64:Darwin:*:*)
+	echo x86_64-apple-darwin${UNAME_RELEASE}
+	exit 0 ;;
     powerpc:machten:*:*)
 	echo powerpc-apple-machten${UNAME_RELEASE}
 	exit 0 ;;
@@ -503,6 +512,9 @@
     *:GNU:*:*)
 	echo `echo ${UNAME_MACHINE}|sed -e 's,[-/].*$,,'`-unknown-gnu`echo ${UNAME_RELEASE}|sed -e 's,/.*$,,'`
 	exit 0 ;;
+    x86_64:Linux:*:*)
+	echo x86_64-unknown-linux-gnu
+	exit 0 ;;
     *:Linux:*:*)
 	# uname on the ARM produces all sorts of strangeness, and we need to
 	# filter it out.
diff -urN src-orig/config.sub src/config.sub
--- src-orig/config.sub	2002-10-22 13:13:56.000000000 +0300
+++ src/config.sub	2014-07-31 10:29:34.000000000 +0300
@@ -222,7 +222,7 @@
 	# We use `pc' rather than `unknown'
 	# because (1) that's what they normally are, and
 	# (2) the word "unknown" tends to confuse beginning users.
-	i[34567]86)
+	i[34567]86 | x86_64)
 	  basic_machine=$basic_machine-pc
 	  ;;
 	# Object if more than one company name word.
@@ -243,7 +243,7 @@
 	      | sparc64-* | sparc86x-* | mips64-* | mipsel-* \
 	      | mips64el-* | mips64orion-* | mips64orionel-*  \
 	      | mipstx39-* | mipstx39el-* \
-	      | f301-*)
+	      | f301-* | x86_64-*)
 		;;
 	m88110-* | m680[01234]0-* | m683?2-* | m68360-* | z8k-* | h8500-* | d10v-*) # CYGNUS LOCAL
 		;;
@@ -955,6 +955,8 @@
 	      | -windows* | -osx | -abug |  -netware* | -os9* | -beos* \
 	      | -macos* | -mpw* | -magic* | -mon960* | -lnews* )
 		;;
+	-darwin*)
+		;;
 	-mac*)
 		os=`echo $os | sed -e 's|mac|macos|'`
 		;;
diff -urN src-orig/configure src/configure
--- src-orig/configure	2002-10-22 13:13:56.000000000 +0300
+++ src/configure	2014-07-31 10:29:34.000000000 +0300
@@ -697,6 +697,7 @@
 if result=`${config_shell} ${configsub} ${host_alias}` ; then
     true
 else
+    echo "${config_shell} ${configsub} ${host_alias}." 1>&2
     echo "Unrecognized host system name ${host_alias}." 1>&2
     exit 1
 fi
@@ -1529,7 +1530,8 @@
 # Perform the same cleanup as the trap handler, minus the "exit 1" of course,
 # and reset the trap handler.
 rm -f ${tmpfile}.com ${tmpfile}.tgt ${tmpfile}.hst ${tmpfile}.pos
-trap 0
+#trap 0
+trap "rm -f Makefile.tem ${tmpfile}.com ${tmpfile}.tgt ${tmpfile}.hst ${tmpfile}.pos; exit 0" 0
 
 exit 0
 
diff -urN src-orig/diff/stamp-h src/diff/stamp-h
--- src-orig/diff/stamp-h	1970-01-01 02:00:00.000000000 +0200
+++ src/diff/stamp-h	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1 @@
+Sat Dec 25 21:46:01 GMT 2004
diff -urN src-orig/diff/system.h src/diff/system.h
--- src-orig/diff/system.h	2002-10-22 13:13:58.000000000 +0300
+++ src/diff/system.h	2014-07-31 10:29:34.000000000 +0300
@@ -93,6 +93,15 @@
 
 #if HAVE_FCNTL_H
 #include <fcntl.h>
+#ifdef HAVE_SETMODE
+#undef HAVE_SETMODE
+#endif
+#ifndef _O_BINARY
+#define _O_BINARY 0
+#endif
+#ifndef O_BINARY
+#define O_BINARY _O_BINARY
+#endif
 #else
 #if HAVE_SYS_FILE_H
 #include <sys/file.h>
diff -urN src-orig/gcc/c-gperf.h src/gcc/c-gperf.h
--- src-orig/gcc/c-gperf.h	2002-10-22 13:14:05.000000000 +0300
+++ src/gcc/c-gperf.h	2014-07-31 10:33:22.000000000 +0300
@@ -165,7 +165,7 @@
 #ifdef __GNUC__
 inline
 #endif
-struct resword *
+static struct resword *
 is_reserved_word (str, len)
      register char *str;
      register unsigned int len;
diff -urN src-orig/gcc/config/arm/arm.h src/gcc/config/arm/arm.h
--- src-orig/gcc/config/arm/arm.h	2002-10-22 13:14:06.000000000 +0300
+++ src/gcc/config/arm/arm.h	2014-07-31 10:29:34.000000000 +0300
@@ -405,7 +405,7 @@
 };
 
 /* Recast the program mode class to be the prog_mode attribute */
-#define arm_prog_mode ((enum attr_prog_mode) arm_prgmode)
+#define arm_prog_mode (/*(enum attr_prog_mode)*/ arm_prgmode)
 
 extern enum prog_mode_type arm_prgmode;
 
diff -urN src-orig/gcc/config/x86_64/xm-x86_64.h src/gcc/config/x86_64/xm-x86_64.h
--- src-orig/gcc/config/x86_64/xm-x86_64.h	1970-01-01 02:00:00.000000000 +0200
+++ src/gcc/config/x86_64/xm-x86_64.h	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,39 @@
+/* Configuration for GNU C-compiler for x86_64
+   Copyright (C) 1988, 1993 Free Software Foundation, Inc.
+
+This file is part of GNU CC.
+
+GNU CC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2, or (at your option)
+any later version.
+
+GNU CC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GNU CC; see the file COPYING.  If not, write to
+the Free Software Foundation, 59 Temple Place - Suite 330,
+Boston, MA 02111-1307, USA.  */
+
+/* #defines that need visibility everywhere.  */
+#define FALSE 0
+#define TRUE 1
+
+/* This describes the machine the compiler is hosted on.  */
+#define HOST_BITS_PER_CHAR 8
+#define HOST_BITS_PER_SHORT 16
+#define HOST_BITS_PER_INT 32
+#define HOST_BITS_PER_LONG 64
+#define HOST_BITS_PER_LONGLONG 64
+
+/* Arguments to use with `exit'.  */
+#define SUCCESS_EXIT_CODE 0
+#define FATAL_EXIT_CODE 33
+
+/* target machine dependencies.
+   tm.h is a symbolic link to the actual target specific file.   */
+
+#include "tm.h"
diff -urN src-orig/gcc/configure src/gcc/configure
--- src-orig/gcc/configure	2002-10-22 13:14:13.000000000 +0300
+++ src/gcc/configure	2014-07-31 10:29:34.000000000 +0300
@@ -4329,6 +4329,15 @@
 		xmake_file=romp/x-mach
 		use_collect2=yes
 		;;
+	powerpc-apple-darwin*)
+		xmake_file=rs6000/x-darwin
+		;;
+	i386-apple-darwin*)
+		xmake_file=i386/x-darwin
+		;;
+	x86_64-apple-darwin*)
+		xmake_file=x86_64/x-darwin
+		;;
 	powerpc-*-beos*)
 		cpu_type=rs6000
 		tm_file=rs6000/beos.h
@@ -5067,6 +5076,9 @@
 		;;
 		# END CYGNUS LOCAL raeburn/vr5400
 
+	x86_64-unknown-linux-gnu)
+		;;
+
 # This hasn't been upgraded to GCC 2.
 	*)
 		echo "Configuration $machine not supported" 1>&2
diff -urN src-orig/gcc/configure.in src/gcc/configure.in
--- src-orig/gcc/configure.in	2002-10-22 13:14:13.000000000 +0300
+++ src/gcc/configure.in	2014-07-31 10:29:34.000000000 +0300
@@ -2327,6 +2327,12 @@
 		xmake_file=romp/x-mach
 		use_collect2=yes
 		;;
+	powerpc-apple-darwin*)
+		xmake_file=rs6000/x-darwin
+		;;
+	i386-apple-darwin*)
+		xmake_file=i386/x-darwin
+		;;
 	powerpc-*-beos*)
 		cpu_type=rs6000
 		tm_file=rs6000/beos.h
@@ -3065,6 +3071,9 @@
 		;;
 		# END CYGNUS LOCAL raeburn/vr5400
 
+	x86_64-unknown-linux-gnu)
+		;;
+
 # This hasn't been upgraded to GCC 2.
 	*)
 		echo "Configuration $machine not supported" 1>&2
diff -urN src-orig/gcc/cp/hash.h src/gcc/cp/hash.h
--- src-orig/gcc/cp/hash.h	2002-10-22 13:14:14.000000000 +0300
+++ src/gcc/cp/hash.h	2014-07-31 10:33:30.000000000 +0300
@@ -57,7 +57,7 @@
 #ifdef __GNUC__
 inline
 #endif
-struct resword *
+static struct resword *
 is_reserved_word (str, len)
      register char *str;
      register unsigned int len;
diff -urN src-orig/gcc/cp/lex.c src/gcc/cp/lex.c
--- src-orig/gcc/cp/lex.c	2002-10-22 13:14:14.000000000 +0300
+++ src/gcc/cp/lex.c	2014-07-31 10:33:46.000000000 +0300
@@ -3126,7 +3126,7 @@
 #ifdef __GNUC__
 __inline
 #endif
-void
+static void
 embedded_pedwarn (s)
      char *s;
 {
diff -urN src-orig/gcc/cplus-dem.c src/gcc/cplus-dem.c
--- src-orig/gcc/cplus-dem.c	2002-10-22 13:14:14.000000000 +0300
+++ src/gcc/cplus-dem.c	2014-07-31 10:29:34.000000000 +0300
@@ -3827,8 +3827,8 @@
   exit (1);
 }
 
-char * malloc ();
-char * realloc ();
+/* char * malloc (); */
+/* char * realloc (); */
 
 char *
 xmalloc (size)
diff -urN src-orig/gcc/obstack.h src/gcc/obstack.h
--- src-orig/gcc/obstack.h	2002-10-22 13:14:19.000000000 +0300
+++ src/gcc/obstack.h	2014-07-31 10:29:34.000000000 +0300
@@ -417,7 +417,8 @@
 ({ struct obstack *__o = (OBSTACK);					\
    if (__o->next_free + sizeof (void *) > __o->chunk_limit)		\
      _obstack_newchunk (__o, sizeof (void *));				\
-   *((void **)__o->next_free)++ = ((void *)datum);			\
+   *((void **)__o->next_free) = (void *) datum;                         \
+   __o->next_free = (char *)((void **)__o->next_free + 1);              \
    (void) 0; })
 
 # define obstack_int_grow(OBSTACK,datum)				\
diff -urN src-orig/gdb/config/i386/macos.mh src/gdb/config/i386/macos.mh
--- src-orig/gdb/config/i386/macos.mh	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/i386/macos.mh	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,3 @@
+
+NATDEPFILES= mac-nat.o
+NAT_FILE= nm-macos.h
diff -urN src-orig/gdb/config/i386/nm-macos.h src/gdb/config/i386/nm-macos.h
--- src-orig/gdb/config/i386/nm-macos.h	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/i386/nm-macos.h	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,20 @@
+/* i386 native-dependent macros for GDB, the GNU debugger.
+   Copyright (C) 1995 Free Software Foundation, Inc.
+
+This file is part of GDB.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+/* Nothing needed, at least so far.  */
diff -urN src-orig/gdb/config/powerpc/xm.h src/gdb/config/powerpc/xm.h
--- src-orig/gdb/config/powerpc/xm.h	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/powerpc/xm.h	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,22 @@
+/* Macro definitions for running GDB on Apple Power Macintoshes.
+   Copyright (C) 1994 Free Software Foundation, Inc.
+
+This file is part of GDB.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+#define HOST_BYTE_ORDER BIG_ENDIAN
+
+//#include <xm-mpw.h>
diff -urN src-orig/gdb/config/x86_64/linux.mh src/gdb/config/x86_64/linux.mh
--- src-orig/gdb/config/x86_64/linux.mh	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/x86_64/linux.mh	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,8 @@
+# Host: x86_64 running GNU/Linux
+
+XM_FILE= xm-linux.h
+XDEPFILES= ser-tcp.o
+
+NAT_FILE= nm-linux.h
+# Stub, i386 files just removed
+NATDEPFILES= infptrace.o solib.o inftarg.o fork-child.o corelow.o core-aout.o core-regset.o
diff -urN src-orig/gdb/config/x86_64/macos.mh src/gdb/config/x86_64/macos.mh
--- src-orig/gdb/config/x86_64/macos.mh	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/x86_64/macos.mh	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,3 @@
+
+NATDEPFILES= mac-nat.o
+NAT_FILE= nm-macos.h
diff -urN src-orig/gdb/config/x86_64/macos.mh.rej src/gdb/config/x86_64/macos.mh.rej
--- src-orig/gdb/config/x86_64/macos.mh.rej	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/x86_64/macos.mh.rej	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,6 @@
+***************
+*** 0 ****
+--- 1,3 ----
++ 
++ NATDEPFILES= mac-nat.o
++ NAT_FILE= nm-macos.h
diff -urN src-orig/gdb/config/x86_64/xm-linux.h src/gdb/config/x86_64/xm-linux.h
--- src-orig/gdb/config/x86_64/xm-linux.h	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/config/x86_64/xm-linux.h	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,36 @@
+/* Native support for GNU/Linux, for GDB, the GNU debugger.
+   Copyright (C) 1986, 1987, 1989, 1992 Free Software Foundation, Inc.
+
+This file is part of GDB.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+#ifndef XM_LINUX_H
+#define XM_LINUX_H
+
+#define HOST_BYTE_ORDER LITTLE_ENDIAN
+
+#define HAVE_TERMIOS
+
+/* This is the amount to subtract from u.u_ar0
+   to get the offset in the core file of the register values.  */
+#define KERNEL_U_ADDR 0x0
+
+#define NEED_POSIX_SETPGID
+
+/* Need R_OK etc, but USG isn't defined.  */
+#include <unistd.h>
+
+#endif	/* #ifndef XM_LINUX_H */
diff -urN src-orig/gdb/configure src/gdb/configure
--- src-orig/gdb/configure	2002-10-22 13:14:36.000000000 +0300
+++ src/gdb/configure	2014-07-31 10:29:34.000000000 +0300
@@ -5586,13 +5586,13 @@
 frags=
 host_makefile_frag=${srcdir}/config/${gdb_host_cpu}/${gdb_host}.mh
 if test ! -f ${host_makefile_frag}; then
-{ echo "configure: error: "*** Gdb does not support host ${host}"" 1>&2; exit 1; }
+{ echo "configure: error: "*** ${srcdir}/config/${gdb_host_cpu}/${gdb_host}.mh Gdb does not support host ${host}"" 1>&2; exit 1; }
 fi
 frags="$frags $host_makefile_frag"
 
 target_makefile_frag=${srcdir}/config/${gdb_target_cpu}/${gdb_target}.mt
 if test ! -f ${target_makefile_frag}; then
-{ echo "configure: error: "*** Gdb does not support target ${target}"" 1>&2; exit 1; }
+{ echo "configure: error: "*** ${srcdir}/config/${gdb_target_cpu}/${gdb_target}.mt Gdb does not support target ${target}"" 1>&2; exit 1; }
 fi
 frags="$frags $target_makefile_frag"
 
diff -urN src-orig/gdb/configure.host src/gdb/configure.host
--- src-orig/gdb/configure.host	2002-10-22 13:14:36.000000000 +0300
+++ src/gdb/configure.host	2014-07-31 10:29:34.000000000 +0300
@@ -28,6 +28,10 @@
 
 case "${host}" in
 
+powerpc-apple-darwin*)		gdb_host=macos ;;
+i386-apple-darwin*)		gdb_host=macos ;;
+x86_64-apple-darwin*)		gdb_host=macos ;;
+
 a29k-*-*)		gdb_host=ultra3 ;;
 
 alpha*-*-osf1*)		gdb_host=alpha-osf1 ;;
@@ -157,4 +161,6 @@
 vax-*-ultrix2*)		gdb_host=vaxult2 ;;
 vax-*-ultrix*)		gdb_host=vaxult ;;
 
+x86_64-*-linux*)	gdb_host=linux ;;
+
 esac
diff -urN src-orig/gdb/rdi-share/configure src/gdb/rdi-share/configure
--- src-orig/gdb/rdi-share/configure	2002-10-22 13:14:39.000000000 +0300
+++ src/gdb/rdi-share/configure	2014-07-31 10:29:34.000000000 +0300
@@ -696,8 +696,9 @@
 
 VERSION=1.0
 
+rm -f $srcdir/config.status
 if test "`cd $srcdir && pwd`" != "`pwd`" && test -f $srcdir/config.status; then
-  { echo "configure: error: source directory already configured; run "make distclean" there first" 1>&2; exit 1; }
+  { echo "configure: dir: [$srcdir] pwd: [`pwd`] error: source directory already configured; run "make distclean" there first" 1>&2; exit 1; }
 fi
 cat >> confdefs.h <<EOF
 #define PACKAGE "$PACKAGE"
diff -urN src-orig/gdb/rdi-share/devsw.c src/gdb/rdi-share/devsw.c
--- src-orig/gdb/rdi-share/devsw.c	2002-10-22 13:14:39.000000000 +0300
+++ src/gdb/rdi-share/devsw.c	2014-07-31 10:29:34.000000000 +0300
@@ -267,7 +267,7 @@
 
     /* Free memory */
     free ((char *) device->SwitcherState);
-    device->SwitcherState = 0x0;
+    ((DeviceDescr*)device)->SwitcherState = 0x0;
 
     /* that's all */
     return adp_ok;
diff -urN src-orig/gdb/rdi-share/hostchan.h src/gdb/rdi-share/hostchan.h
--- src-orig/gdb/rdi-share/hostchan.h	2002-10-22 13:14:39.000000000 +0300
+++ src/gdb/rdi-share/hostchan.h	2014-07-31 10:29:34.000000000 +0300
@@ -16,6 +16,9 @@
 #define angsd_hostchan_h
 
 /* struct timeval */
+#ifndef __unix
+#define __unix
+#endif
 #if defined(__unix) || defined(__CYGWIN32__)
 #  include <sys/time.h>
 #else
diff -urN src-orig/gdb/rdi-share/unixcomm.c src/gdb/rdi-share/unixcomm.c
--- src-orig/gdb/rdi-share/unixcomm.c	2002-10-22 13:14:39.000000000 +0300
+++ src/gdb/rdi-share/unixcomm.c	2014-07-31 10:29:34.000000000 +0300
@@ -98,6 +98,13 @@
 #define SERIAL_PREFIX "com"
 #endif
 
+#ifndef SERPORT1
+#define SERPORT1   "/dev/ttya"
+#define SERPORT2   "/dev/ttyb"
+#define PARPORT1   "/dev/bpp0"
+#define PARPORT2   "/dev/bpp1"
+#endif
+
 /*
  * Parallel port output pins, used for signalling to target
  */
diff -urN src-orig/gdb/remote-rdp.c src/gdb/remote-rdp.c
--- src-orig/gdb/remote-rdp.c	2002-10-22 13:14:39.000000000 +0300
+++ src/gdb/remote-rdp.c	2014-07-31 10:29:34.000000000 +0300
@@ -490,7 +490,7 @@
 	  abort ();
 	}
     }
-  va_end (args);
+  va_end (alist);
 
   if (dst != buf)
     abort ();
diff -urN src-orig/gdb/xm.h src/gdb/xm.h
--- src-orig/gdb/xm.h	1970-01-01 02:00:00.000000000 +0200
+++ src/gdb/xm.h	2014-07-31 10:29:34.000000000 +0300
@@ -0,0 +1,26 @@
+/* Macro definitions for running GDB on Apple Power Macintoshes.
+   Copyright (C) 1994 Free Software Foundation, Inc.
+
+This file is part of GDB.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+#define HOST_BYTE_ORDER BIG_ENDIAN
+
+#ifndef __unix
+#define __unix
+#endif
+
+//#include <xm-mpw.h>
diff -urN src-orig/libiberty/cplus-dem.c src/libiberty/cplus-dem.c
--- src-orig/libiberty/cplus-dem.c	2002-10-22 13:14:49.000000000 +0300
+++ src/libiberty/cplus-dem.c	2014-07-31 10:29:34.000000000 +0300
@@ -3827,8 +3827,8 @@
   exit (1);
 }
 
-char * malloc ();
-char * realloc ();
+/* char * malloc (); */
+/* char * realloc (); */
 
 char *
 xmalloc (size)
diff -urN src-orig/libiberty/strerror.c src/libiberty/strerror.c
--- src-orig/libiberty/strerror.c	2002-10-22 13:14:49.000000000 +0300
+++ src/libiberty/strerror.c	2014-07-31 10:29:34.000000000 +0300
@@ -13,6 +13,7 @@
    incompatible with our later declaration, perhaps by using const
    attributes.  So we hide the declaration in errno.h (if any) using a
    macro. */
+#define sys_nerr sys_nerr__
 #define sys_errlist sys_errlist__
 #endif
 
@@ -20,6 +21,7 @@
 #include <errno.h>
 
 #ifdef HAVE_SYS_ERRLIST
+#undef sys_nerr
 #undef sys_errlist
 #endif
 
diff -urN src-orig/libiberty/strsignal.c src/libiberty/strsignal.c
--- src-orig/libiberty/strsignal.c	2002-10-22 13:14:49.000000000 +0300
+++ src/libiberty/strsignal.c	2014-07-31 10:29:34.000000000 +0300
@@ -17,6 +17,7 @@
    ignore any declaration in the system header files, and always
    declare it ourselves.  With luck, this will always work.  */
 #define sys_siglist no_such_symbol
+#define sys_nsig sys_nsig__no_such_symbol
 
 #include <stdio.h>
 #include <signal.h>
@@ -35,6 +36,7 @@
 /* Undefine the macro we used to hide the definition of sys_siglist
    found in the system header files.  */
 #undef sys_siglist
+#undef sys_nsig
 
 #ifndef NULL
 #  ifdef __STDC__
diff -urN src-orig/libstdc++/stdexcepti.cc src/libstdc++/stdexcepti.cc
--- src-orig/libstdc++/stdexcepti.cc	2002-10-22 13:14:50.000000000 +0300
+++ src/libstdc++/stdexcepti.cc	2014-07-31 10:29:34.000000000 +0300
@@ -12,10 +12,10 @@
 void
 __out_of_range (const char *s)
 {
-  throw out_of_range (s);
+ // throw out_of_range (s);
 }
 
 void __length_error (const char *s)
 {
-  throw length_error (s);
+  //throw length_error (s);
 }
diff -urN src-orig/mmalloc/mmap-sup.c src/mmalloc/mmap-sup.c
--- src-orig/mmalloc/mmap-sup.c	2002-10-22 13:14:52.000000000 +0300
+++ src/mmalloc/mmap-sup.c	2014-07-31 10:29:34.000000000 +0300
@@ -31,6 +31,7 @@
 #endif
 
 #include "mmprivate.h"
+#include <sys/types.h>
 
 /* Cache the pagesize for the current host machine.  Note that if the host
    does not readily provide a getpagesize() function, we need to emulate it
diff -urN src-orig/tcl/unix/configure src/tcl/unix/configure
--- src-orig/tcl/unix/configure	2002-10-22 13:15:07.000000000 +0300
+++ src/tcl/unix/configure	2014-07-31 10:29:34.000000000 +0300
@@ -4170,9 +4170,9 @@
 	# Special check for weird MP-RAS system (uname returns weird
 	# results, and the version is kept in special file).
     
-	if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-	    system=MP-RAS-`awk '{print $3}' /etc/.relid'`
-	fi
+#	if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
+#	    system=MP-RAS-`awk '{print $3}' /etc/.relid'`
+#	fi
 	if test "`uname -s`" = "AIX" ; then
 	    system=AIX-`uname -v`.`uname -r`
 	fi
@@ -5072,9 +5072,9 @@
 	# Special check for weird MP-RAS system (uname returns weird
 	# results, and the version is kept in special file).
     
-	if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-	    system=MP-RAS-`awk '{print $3}' /etc/.relid'`
-	fi
+#	if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
+#	    system=MP-RAS-`awk '{print $3}' /etc/.relid'`
+#	fi
 	if test "`uname -s`" = "AIX" ; then
 	    system=AIX-`uname -v`.`uname -r`
 	fi
diff -urN src-orig/texinfo/configure src/texinfo/configure
--- src-orig/texinfo/configure	2002-10-22 13:15:08.000000000 +0300
+++ src/texinfo/configure	2014-07-31 10:29:34.000000000 +0300
@@ -1339,29 +1339,31 @@
 test "$am_cv_mingw32" = yes && MINGW32=yes
 
 
-echo $ac_n "checking for executable suffix""... $ac_c" 1>&6
-echo "configure:1344: checking for executable suffix" >&5
-if eval "test \"`echo '$''{'am_cv_exeext'+set}'`\" = set"; then
-  echo $ac_n "(cached) $ac_c" 1>&6
-else
-  if test "$CYGWIN32" = yes || test "$MINGW32" = yes; then
-am_cv_exeext=.exe
-else
-cat > am_c_test.c << 'EOF'
-int main() {
-/* Nothing needed here */
-}
-EOF
-${CC-cc} -o am_c_test $CFLAGS $CPPFLAGS $LDFLAGS am_c_test.c $LIBS 1>&5
-am_cv_exeext=`echo am_c_test.* | grep -v am_c_test.c | sed -e s/am_c_test//`
-rm -f am_c_test*
-fi
-
-test x"${am_cv_exeext}" = x && am_cv_exeext=no
-fi
+#echo $ac_n "checking for executable suffix""... $ac_c" 1>&6
+#echo "configure:1344: checking for executable suffix" >&5
+#if eval "test \"`echo '$''{'am_cv_exeext'+set}'`\" = set"; then
+#  echo $ac_n "(cached) $ac_c" 1>&6
+#else
+#  if test "$CYGWIN32" = yes || test "$MINGW32" = yes; then
+#am_cv_exeext=.exe
+#else
+#cat > am_c_test.c << 'EOF'
+#int main() {
+#/* Nothing needed here */
+#}
+#EOF
+#${CC-cc} -o am_c_test $CFLAGS $CPPFLAGS $LDFLAGS am_c_test.c $LIBS 1>&5
+#am_cv_exeext=`echo am_c_test.* | grep -v am_c_test.c | sed -e s/am_c_test//`
+#rm -f am_c_test*
+#fi
+#
+#test x"${am_cv_exeext}" = x && am_cv_exeext=no
+#fi
+#EXEEXT=""
+#test x"${am_cv_exeext}" != xno && EXEEXT=${am_cv_exeext}
+#echo "$ac_t""${am_cv_exeext}" 1>&6
 EXEEXT=""
-test x"${am_cv_exeext}" != xno && EXEEXT=${am_cv_exeext}
-echo "$ac_t""${am_cv_exeext}" 1>&6
+am_cv_exeext=""
 
 
 
